include/magma/vk/pipeline/PipelineInfo.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+#include <vulkan/vulkan.hpp>
+#include <vector>
+
+class PipelineLayoutInfo {
+public:
+    PipelineLayoutInfo();
+    PipelineLayoutInfo(const std::vector<vk::DescriptorSetLayout> &descriptorSetLayouts);
+    PipelineLayoutInfo(const vk::DescriptorSetLayout &descriptorSetLayout);
+
+    ~PipelineLayoutInfo() = default;
+
+    const vk::PipelineLayoutCreateInfo &getPipelineLayoutInfo() const { return _pipelineLayoutInfo; }
+
+private:
```

Comment:
It's better to save as private member std::vector<vk::DescriptorSetLayout> from which object is constructed to prevent creation pointers to dead memory.